File: assignment-3-crud-practice-Jwizzed-main/Person.py
import math
import re


class Person:
    def __init__(self, name, idNumber, contactNumber):
        self.name = name
        self.idNumber = idNumber
        self.contactNumber = contactNumber

    @property
    def name(self):
        return self.__name

    @name.setter
    def name(self, name):
        self.__name = name

    @property
    def idNumber(self):
        return self.__idNumber

    @idNumber.setter
    def idNumber(self, idNumber):

        new_num = idNumber
        while new_num > 10:
            new_num /= 10

        if int(math.log10(idNumber)) == 7 and (new_num % 10 != 0):
            self.__idNumber = idNumber
        else:
            raise ValueError("Invalid ID number\n")

    @property
    def contactNumber(self):
        return self.__contactNumber

    @contactNumber.setter
    def contactNumber(self, contactNumber):
        contactNumber = contactNumber.replace("[^0-9]", "")
        pattern = re.compile("^02\d{7}|0\d{9}", re.IGNORECASE)
        matcher = pattern.match(contactNumber)
        if matcher:
            self.__contactNumber = contactNumber
        else:
            raise ValueError("Invalid contact number\n")

    def __str__(self):
        return "Records{" + "name=" + self.name + ", idNumber=" + str(self.idNumber) + ", contactNumber=" + self.contactNumber + '}'

File: assignment-5-Jwizzed-main/Person.cs
namespace ConsoleApplication1;

public class Person
{
    private string name;
    private int idNumber;
    private string contactNumber;

    public Person()
    {
    }

    public Person(String name, int idNumber, String contactNumber)
    {
        setName(name);
        setIdNumber(idNumber);
        setContactNumber(contactNumber);
    }

    public string getName()
    {
        return name;
    }

    public void setName(string name)
    {
        this.name = name;
    }

    public int getIdNumber()
    {
        return idNumber;
    }

    public void setIdNumber(int idNumber)
    {
        int new_num = idNumber;
        while (new_num > 10)
        {
            new_num /= 10;
        }

        if ((int)Math.Log10(idNumber) == 7 && (new_num % 10 != 0))
        {
            this.idNumber = idNumber;
        }
        else
        {
            throw new Exception(("Invalid ID number"));
        }
    }
    public String getContactNumber() {
        return contactNumber;
    }
    public void setContactNumber(String contactNumber) {
        contactNumber = contactNumber.Replace("-","").Replace("(","").Replace(")","");
        if(contactNumber.StartsWith("02") && (contactNumber.Length == 9)){
            this.contactNumber = contactNumber;
        }else if(contactNumber.StartsWith("0") && (contactNumber.Length == 10)){
            this.contactNumber = contactNumber;
        }else{
            throw new Exception(("Invalid contact number"));
        }
    }

    public String toString(){
        return "Records{" + "name=" + name + ", idNumber=" + idNumber + ", contactNumber=" + contactNumber + '}';
    }
}